agent/lib_aws_sdk_php.c

```diff
@@ -295,6 +302,166 @@ void nr_lib_aws_sdk_php_sqs_parse_queueurl(
   cloud_attrs->cloud_region = region;
 }
 
+void nr_lib_aws_sdk_php_lambda_handle(nr_segment_t* auto_segment,
+                                      char* command_name_string,
+                                      size_t command_name_len,
+                                      NR_EXECUTE_PROTO) {
+  nr_segment_t* external_segment = NULL;
+  zval** retval_ptr = NR_GET_RETURN_VALUE_PTR;
+
+  nr_segment_cloud_attrs_t cloud_attrs = {
+      .cloud_platform = "aws_lambda"
+  };
+
+  if (NULL == auto_segment) {
+    return;
+  }
+
+  if (NULL == command_name_string || 0 == command_name_len) {
+    return;
+  }
+
+  if (NULL == retval_ptr) {
+    /* Do not instrument when an exception has happened */
+    return;
+  }
+
+#define AWS_COMMAND_IS(CMD) \
+  (command_name_len == (sizeof(CMD) - 1) && nr_streq(CMD, command_name_string))
+
+  /* Determine if we instrument this command. */
+  if (AWS_COMMAND_IS("invoke")) {
+    // Command can be saved here if in the future
+    // we instrument more than 1 lambda command
+  } else {
+    return;
+  }
+#undef AWS_COMMAND_IS
+
+  /* reconstruct the ARN */
+  nr_aws_lambda_invoke(NR_EXECUTE_ORIG_ARGS, &cloud_attrs);
+  if (!cloud_attrs.cloud_resource_id) {
+    /* we do not want to instrument if we cannot reconstruct the ARN */
+    return;
```

Comment:
I think it's still fine to instrument in this case.
It won't create the relationship, but the external segment can still be created.  The cloud attributes are extra attributes, not core external segment attributes.